transaction/transaction.go

```diff
@@ -358,9 +366,187 @@ func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetTransferTxn(account, recipient, closeAssetsTo string, amount, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	var tx types.Transaction
+
+	tx.Type = types.AssetTransferTx
+
+	accountAddr, err := types.DecodeAddress(account)
+	if err != nil {
+		return tx, err
+	}
+
+	ghBytes, err := byte32FromBase64(genesisHash)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Header = types.Header{
+		Sender:      accountAddr,
+		Fee:         types.MicroAlgos(feePerByte),
+		FirstValid:  types.Round(firstRound),
+		LastValid:   types.Round(lastRound),
+		GenesisHash: types.Digest(ghBytes),
+		GenesisID:   genesisID,
+		Note:        note,
+	}
+
+	creatorAddr, err := types.DecodeAddress(creator)
+	if err != nil {
+		return tx, err
+	}
+
+	tx.XferAsset = types.AssetID{
+		Creator: creatorAddr,
+		Index:   index,
+	}
+
+	recipientAddr, err := types.DecodeAddress(recipient)
+	if err != nil {
+		return tx, err
+	}
+	tx.AssetReceiver = recipientAddr
+
+	if closeAssetsTo != "" {
+		closeToAddr, err := types.DecodeAddress(closeAssetsTo)
+		if err != nil {
+			return tx, err
+		}
+		tx.AssetCloseTo = closeToAddr
+	}
+
+	tx.AssetAmount = amount
+
+	// Update fee
+	eSize, err := estimateSize(tx)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+	tx.Fee = types.MicroAlgos(eSize * feePerByte)
+
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+
 	return tx, nil
 }
 
+// MakeAssetAcceptanceTransaction creates a tx for marking an account as willing to accept the given asset
+// - account is a checksummed, human-readable address that will send the transaction and begin accepting the asset
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetAcceptanceTxn(account string, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetTransferTxn(account, account, "", 0,
+		feePerByte, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+	return tx, err
+}
+
+// MakeAssetRevocationTransaction creates a tx for revoking an asset from an account and sending it to another
```

Comment:
Do we actually need this function? Practically, just the revocation manager can revoke assets so it seems a bit redundant to the regular transfer asset function

Reply:
@rotemh I think it's worth it to keep the revocation helper, since it adds handling `tx.AssetSend`/`asnd`, which normal transfer doesn't need to worry about. I agree that `Mint` or `Burn` don't add anything, I'll remove them.

Reply:
Follow-up: nevermind, leaving them for now per our in-person conversation